stm32/.gitignore

```diff
@@ -3,6 +3,9 @@
 .vscode/c_cpp_properties.json
 .vscode/launch.json
 .vscode/ipch
+.vscode/settings.json
 
 # docs
-docs/html
\ No newline at end of file
+docs/html
+
+stm32pio.ini
```

Comment:
I like this, I think we just make a template and turn the file into a per user configuration. We should provide a template file and `stm32pio.ini` should be removed from git.